From 570bde7ac6cc4eb5d020449c424ba3a85ebe2678 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Sat, 28 Jul 2012 00:30:41 +0800
Subject: NuPlayerRender: Fix uint32_t numFramesPendingPlayout overrun issue

BZ: 48710

Add protection to avoid uint32_t numFramesPendingPlayout overrun.
If overrun happens, the video will be post a large delay time casuing video frozen.

orignal patch
43855: orig-bz(BZ: 27202)

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: I81900b72e45006c1584ad5b0845c33922b5dcb4b
Orig-Change-Id: I9e9bb8c90c4c12f35f49857b69c70c7078a01b05
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 .../nuplayer/NuPlayerRenderer.cpp                  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
index f3e7fd7..c53330b 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
@@ -285,7 +285,7 @@ bool NuPlayer::Renderer::onDrainAudioQueue() {
             CHECK_EQ(mAudioSink->getPosition(&numFramesPlayed), (status_t)OK);
 
             uint32_t numFramesPendingPlayout =
-                mNumFramesWritten - numFramesPlayed;
+                (mNumFramesWritten > numFramesPlayed) ? (mNumFramesWritten - numFramesPlayed) : 0;
 
             int64_t realTimeOffsetUs =
                 (mAudioSink->latency() / 2  /* XXX */
-- 
1.7.4.1

